pkg/server/api/entry/v1/service.go

```diff
@@ -168,24 +168,32 @@ func (s *Service) createEntry(ctx context.Context, e *types.Entry, outputMask *t
 
 	log = log.WithField(telemetry.SPIFFEID, cEntry.SpiffeId)
 
-	// Validates that there is no similar entry
-	if isUniqueStatus := s.isEntryUnique(ctx, cEntry, log); isUniqueStatus != nil {
+	existingEntry, err := s.getExistingEntry(ctx, cEntry)
+	if err != nil {
 		return &entry.BatchCreateEntryResponse_Result{
-			Status: isUniqueStatus,
+			Status: api.MakeStatus(log, codes.Internal, "failed to list entries", err),
 		}
 	}
 
-	// Create entry
-	resp, err := s.ds.CreateRegistrationEntry(ctx, &datastore.CreateRegistrationEntryRequest{
-		Entry: cEntry,
-	})
-	if err != nil {
-		return &entry.BatchCreateEntryResponse_Result{
-			Status: api.MakeStatus(log, codes.Internal, "failed to create entry", err),
+	resultStatus := api.OK()
+	regEntry := existingEntry
+
+	if existingEntry == nil {
+		// Create entry
+		resp, err := s.ds.CreateRegistrationEntry(ctx, &datastore.CreateRegistrationEntryRequest{
+			Entry: cEntry,
+		})
+		if err != nil {
+			return &entry.BatchCreateEntryResponse_Result{
+				Status: api.MakeStatus(log, codes.Internal, "failed to create entry", err),
+			}
 		}
+		regEntry = resp.Entry
+	} else {
+		resultStatus = api.MakeStatus(log, codes.AlreadyExists, "entry already exists", nil)
```

Comment:
Hmm, the use of `api.MakeStatus` causes an error log to be emitted. Considering this is a normal use case and not an error, I wonder if should use `api.CreateStatus` here so we don't get the logging.

Reply:
True, I was trying to avoid "breaking" the test that checks the log message, but no one should be relying on that log message. Updated (hopefully.)